### PR TITLE
Fix depwarns on julia v0.5

### PR DESCRIPTION
**Pull request by nkottary**
Fix some deprecation warnings on julia v0.5

### CODE REVIEW

**Comment by tkelman** [comment]
Update the minimum needed version of Compat in REQUIRE

File: src/BinDeps.jl [hunk 1]
```diff
@@ -412,24 +414,32 @@ module BinDeps
...
 
-    	@windows_only @dependent_steps begin
-    		begin
-                ChangeDirectory(s.src)
-    			@windows_only FileRule(isempty(s.config_status_dir)?"config.status":joinpath(s.config_status_dir,"config.status"),setenv(`sh -c $cmdstring`,env))
-                FileRule(s.libtarget,MakeTargets())
-                MakeTargets("install")
+        @static if is_windows() 
+            @dependent_steps begin
+                begin
```
Comment (tkelman) [inline src/BinDeps.jl]: this seems duplicated and unnecessary?

File: src/dependencies.jl [hunk 2]
```diff
@@ -278,7 +278,7 @@ end
 
 lower(x::GetSources,collection) = push!(collection,generate_steps(x.dep,gethelper(x.dep,Sources)...))
 
-Autotools(;opts...) = Autotools(nothing, Dict{Any,Any}([k => v for (k,v) in opts]))
+Autotools(;opts...) = Autotools(nothing, (Any,Any)[k => v for (k,v) in opts])
```
Comment (tkelman) [inline src/dependencies.jl]: This should still be a dictionary

File: src/dependencies.jl [hunk 3]
```diff
@@ -603,19 +603,19 @@ function check_system_handle!(ret,dep,handle)
...
+elseif is_windows()
     defaults = [Binaries,PackageManager,SystemPaths]
 else
     defaults = [SystemPaths,BuildProcess]
 end
 
 function applicable(dep::LibraryDependency)
     if haskey(dep.properties,:os)
-        if (dep.properties[:os] != OS_NAME && dep.properties[:os] != :Unix) || (dep.properties[:os] == :Unix && !Base.is_unix(OS_NAME))
+        if (dep.properties[:os] != Sys.KERNEL && dep.properties[:os] != :Unix) || (dep.properties[:os] == :Unix && !Base.is_unix(OS_NAME))
```
Comment (tkelman) [inline src/dependencies.jl]: this will break on Windows

File: test/testscripts/simplebuild/build.jl [hunk 4]
```diff
@@ -26,4 +26,4 @@ deps = [
 provides(Sources,testuri,liba,SHA="769c43644f239d8825cefc998124060cf9f477f94e8e338f6c3e17839470229d")
 provides(BuildProcess,Autotools(libtarget = "liba.$shlib_ext"),liba)
 
-@BinDeps.install [:liba => :jl_liba]
+@BinDeps.install Dict(:liba => :jl_liba)
```
Comment (tkelman) [inline test/testscripts/simplebuild/build.jl]: needs `@compat`

File: src/BinDeps.jl [hunk 5]
```diff
@@ -361,26 +361,28 @@ module BinDeps
...
     end
 
-    @unix_only function lower(a::MakeTargets,collection)
-        cmd = `make -j8`
-        if(!isempty(a.dir))
-            cmd = `$cmd -C $(a.dir)`
-        end
-        if(!isempty(a.targets))
-            cmd = `$cmd $(a.targets)`
+    @static if is_unix()
```
Comment (yuyichao) [inline src/BinDeps.jl]: `@static` is unnecessary

File: src/BinDeps.jl [hunk 6]
```diff
@@ -361,26 +361,28 @@ module BinDeps
...
     lower(s::SynchronousStepCollection,collection) = (collection|=s)
 
     lower(s) = (c=SynchronousStepCollection();lower(s,c);c)
 
     #run(s::MakeTargets) = run(@make_steps (s,))
 
     function lower(s::AutotoolsDependency,collection)
-    	@windows_only prefix = replace(replace(s.prefix,"\\","/"),"C:/","/c/")
-    	@unix_only prefix = s.prefix
+        is_windows() && (prefix = replace(replace(s.prefix,"\\","/"),"C:/","/c/"))
```
Comment (yuyichao) [inline src/BinDeps.jl]: You might want `@static` here

File: src/BinDeps.jl [hunk 7]
```diff
@@ -412,24 +414,32 @@ module BinDeps
...
-                ChangeDirectory(s.builddir)
-                @unix_only FileRule(isempty(s.config_status_dir)?"config.status":joinpath(s.config_status_dir,"config.status"), setenv(`$(s.src)/configure $(s.configure_options) --prefix=$(prefix)`,env))
-                FileRule(s.libtarget,MakeTargets(;env=s.env))
-                MakeTargets("install";env=env)
+        @static if is_unix()
+            @dependent_steps begin
+                CreateDirectory(s.builddir)
+                begin
+                    ChangeDirectory(s.builddir)
+                    @static if is_unix()
```
Comment (yuyichao) [inline src/BinDeps.jl]: Duplicated here too

File: src/dependencies.jl [hunk 8]
```diff
@@ -543,7 +543,7 @@ function _find_library(dep::LibraryDependency; provider = Any)
             opath = string(lib,ext)
             check_path!(ret,dep,opath)
         end
-        @linux_only begin
```
Comment (yuyichao) [inline src/dependencies.jl]: The ones in this function (4 above including this one) should have `@static`

**Comment by nkottary** [comment]
Modified the code as per the above suggestions.

**Comment by ararslan** [comment]
Bump. @nkottary, can you resolve the branch conflicts? It'd be great to get this merged.

File: src/BinDeps.jl [hunk 9]
```diff
@@ -361,26 +361,32 @@ module BinDeps
...
 
     #run(s::MakeTargets) = run(@make_steps (s,))
 
     function lower(s::AutotoolsDependency,collection)
-    	@windows_only prefix = replace(replace(s.prefix,"\\","/"),"C:/","/c/")
-    	@unix_only prefix = s.prefix
+        @static if is_windows()
+            prefix = replace(replace(s.prefix,"\\","/"),"C:/","/c/")
+        end
+        @static if is_unix()
```
Comment (tkelman) [inline src/BinDeps.jl]: static not needed (or right above)

File: src/dependencies.jl [hunk 10]
```diff
@@ -1,5 +1,4 @@
 # This is the high level interface for building dependencies using the declarative BinDeps Interface
-
```
Comment (tkelman) [inline src/dependencies.jl]: avoid making unrelated changes if you can

File: src/dependencies.jl [hunk 11]
```diff
@@ -239,7 +238,7 @@ immutable SystemPaths <: DependencyProvider; end
 
 show(io::IO, ::SystemPaths) = print(io,"System Paths")
 
-using URIParser
+using URIParser, Compat
```
Comment (tkelman) [inline src/dependencies.jl]: you still need to update the minimum version

File: src/dependencies.jl [hunk 12]
```diff
@@ -468,8 +467,12 @@ function generate_steps(dep::LibraryDependency, h::Autotools,  provider_opts)
     env = Dict{String,String}()
     env["PKG_CONFIG_PATH"] = join(opts[:pkg_config_dirs],":")
     delete!(opts,:pkg_config_dirs)
-    @unix_only env["PATH"] = bindir(dep)*":"*ENV["PATH"]
-    @windows_only env["PATH"] = bindir(dep)*";"*ENV["PATH"]
+    @static if is_unix()
```
Comment (tkelman) [inline src/dependencies.jl]: static not needed (or right below)

File: src/dependencies.jl [hunk 13]
```diff
@@ -468,8 +467,12 @@ function generate_steps(dep::LibraryDependency, h::Autotools,  provider_opts)
...
+        env["PATH"] = bindir(dep)*":"*ENV["PATH"]
+    end
+    @static if is_windows()
```
Comment (tkelman) [inline src/dependencies.jl]: combine into the previous line with `elseif`

File: src/dependencies.jl [hunk 14]
```diff
@@ -508,7 +511,9 @@ function _find_library(dep::LibraryDependency; provider = Any)
         end
 
         # Windows, do you know what `lib` stands for???
-        @windows_only push!(paths,bindir(p,dep))
+        @static if is_windows()
```
Comment (tkelman) [inline src/dependencies.jl]: static not needed

File: src/BinDeps.jl (hunk 1 above)
Comment (tkelman) [inline src/BinDeps.jl]: there are two begins here, shouldn't need both

**Comment by nkottary** [comment]
Modified the code as per above suggestions.

File: src/dependencies.jl (hunk 8 above)
Comment (ararslan) [inline src/dependencies.jl]: Looks like you're still missing the `@static`s here, @nkottary.

**Comment by ararslan** [comment]
This is exciting. Thanks so much for working on this!

File: src/dependencies.jl (hunk 8 above)
Comment (tkelman) [inline src/dependencies.jl]: i think only the ones that use ccall are needed

File: src/BinDeps.jl [hunk 15]
```diff
@@ -364,26 +364,31 @@ module BinDeps
...
 
     function lower(s::AutotoolsDependency,collection)
-    	@windows_only prefix = replace(replace(s.prefix,"\\","/"),"C:/","/c/")
-    	@unix_only prefix = s.prefix
+        if is_windows()
+            prefix = replace(replace(s.prefix,"\\","/"),"C:/","/c/")
+        elseif is_unix()
+            prefix = s.prefix
+        end
     	cmdstring = "pwd && ./configure --prefix=$(prefix) "*join(s.configure_options," ")
```
Comment (tkelman) [inline src/BinDeps.jl]: this should be tested on a package that uses AutotoolsDependency, but you likely need to restructure the above so `prefix = s.prefix` is set outside the if, then only on windows it gets overwritten

Comment (ararslan) [inline src/BinDeps.jl]: Is that the last change needed before this can get merged?

**Comment by nkottary** [comment]
Modified the code as per @tkelman 's last comment.

**Comment by ararslan** [comment]
Party time! :confetti_ball: Thanks again for working on this, @nkottary. Can we tag a new version now that this has been merged?

**Comment by blakejohnson** [comment]
Sure, I'll tag a new version.
